programs/castle-vault/src/adapters/port.rs

```diff
@@ -168,3 +169,70 @@ impl ReserveAccessor for Reserve {
         Ok(reserve)
     }
 }
+
+#[derive(Accounts)]
+pub struct PortRefresher<'info> {
+    /// Vault state account
+    /// Checks that the accounts passed in are correct
+    #[account(
+        mut,
+        has_one = vault_port_lp_token,
+        has_one = port_reserve,
+    )]
+    pub vault: Box<Account<'info, Vault>>,
+
+    /// Token account for the vault's port lp tokens
+    pub vault_port_lp_token: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        executable,
+        address = port_lending_id(),
+    )]
+    pub port_program: AccountInfo<'info>,
+
+    #[account(mut)]
+    pub port_reserve: Box<Account<'info, PortReserve>>,
+
+    //#[soteria(ignore)]
+    pub port_oracle: AccountInfo<'info>,
+
+    pub clock: Sysvar<'info, Clock>,
+}
+
+impl<'info> PortRefresher<'info> {
+    fn port_refresh_reserve_context(
+        &self,
+        use_oracle: bool,
+    ) -> CpiContext<'_, '_, '_, 'info, port_anchor_adaptor::RefreshReserve<'info>> {
+        let oracle_vec = if use_oracle {
+            vec![self.port_oracle.clone()]
+        } else {
+            vec![]
+        };
+        CpiContext::new(
+            self.port_program.clone(),
+            port_anchor_adaptor::RefreshReserve {
+                reserve: self.port_reserve.to_account_info(),
+                clock: self.clock.to_account_info(),
+            },
+        )
+        .with_remaining_accounts(oracle_vec)
+    }
+}
+
+impl<'info> Refresher for PortRefresher<'info> {
+    fn update_actual_allocation(&mut self, use_oracle: bool) -> ProgramResult {
```

Comment:
I'm not sure about this. `remaining_accounts` is for fee collection, and is not related to `use_oracle`.

Reply:
`remaining accounts` in the `ConsolidateRefresh` is used for fee collection. But since that is a separate IX now, we can use `remaining_accounts` in `RefreshPort` for the optional oracle and remove `use_oracle` parameter.

Reply:
Ah I see what you mean. Good idea, I'll update it.